pkg/kubernetes/controllerutil.go

```diff
@@ -151,12 +151,12 @@ func mutate(f controllerutil.MutateFn, key client.ObjectKey, obj runtime.Object)
 	return nil
 }
 
-// LookupResource lookup for resource to be created by separate resource controller
-func LookupResource(name, namespace string, resource runtime.Object, client client.Client, logger logr.Logger) (*reconcile.Result, error) {
-	err := client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: name}, resource)
+// LookupResourceForController lookup for resource to be created by separate resource controller
+func LookupResource(name, namespace string, resource runtime.Object, cwe eventlog.ControllerWithEvents) (*reconcile.Result, error) {
```

Comment:
I think we should have an option, to skip sending events for some cases

Reply:
I've added an optional bool arg so user can disable event in func call